### ISSUE TITLE
perf(jax): Add jax[cuda] optional dependency for GPU acceleration

### ISSUE THREAD

**Issue by ericchansen** [ericchansen/q2mm#150]
## Summary

Add a `gpu` (or `jax-cuda`) optional dependency extra to `pyproject.toml` that installs `jax[cuda12]` instead of plain `jax`. This enables GPU acceleration for both `JaxEngine` and `JaxMDEngine` with **zero code changes** — JAX auto-dispatches to GPU when a CUDA backend is available.

## Motivation

Both JAX-based engines already use `jax.jit`, `jax.grad`, `jax.hessian`, and `jax.vmap` — all of which run on GPU transparently. The only thing preventing GPU use is that `pyproject.toml` lists plain `jax` and `jaxlib` as dependencies, which installs CPU-only builds.

JAX on GPU provides:
- Faster JIT-compiled energy evaluations
- GPU-accelerated automatic differentiation (analytical Hessians, parameter gradients)
- `vmap`-parallelized bonded geometry calculations (already used in JaxMDEngine)

## Proposed Changes

1. **New optional extra** in `pyproject.toml`:
   ```toml
   [project.optional-dependencies]
   gpu = ["jax[cuda12]"]
   ```
2. **Update `all` extra** to note that GPU requires separate install (since `jax[cuda12]` conflicts with plain `jax`)
3. **Documentation** — add a "GPU acceleration" section to docs explaining how to install with GPU support
4. **Optional: device logging** — add a one-liner in the JAX engine constructors to log `jax.devices()` so users can confirm GPU is active

## Effort

Minimal — packaging change + docs. No code changes to engine logic.
